pkg/datapath/linux/node.go

```diff
@@ -946,9 +951,246 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *nodeTypes.Node, firstAdd
 		}
 	}
 
+	if option.Config.IPAM == ipamOption.IPAMENI {
+		addedInterfaces, removedInterfaces := diffInterfaces(oldNode.Interfaces, newNode.Interfaces)
+
+		// Configure new interfaces.
+		for _, addedInterface := range addedInterfaces {
+			mtu := 1500 // FIXME where do we get the MTU from?
+			if _, err := linuxrouting.RetrieveIfIndexFromMAC(addedInterface.MAC, mtu); err != nil {
+				return err
+			}
+		}
+
+		// Ignore removed interfaces for now.
+		_ = removedInterfaces
+
+		oldRules, oldRoutes := nodeRulesAndRoutes(oldNode)
+		newRules, newRoutes := nodeRulesAndRoutes(newNode)
+
+		addedRules, removedRules := diffRules(oldRules, newRules)
+		for _, addedRule := range addedRules {
+			if err := route.ReplaceRule(*addedRule); err != nil {
+				return fmt.Errorf("unable to install ip rule: %s", err)
```

Comment:
Does it make sense to error out here? We should probably attempt to add as many rules as possible instead and log errors. The node handler erroring out doesn't really block anything. This *must* succeed so we should probably rather retry instead.